packages/backend/src/services/PromoteService/PromoteService.ts

```diff
@@ -811,16 +847,24 @@ export class PromoteService extends BaseService {
         };
     }
 
-    async createNewSpaces(
+    async upsertSpaces(
```

Comment:
now we can detect `name` changes in spaces, so we can also   update spaces if needed